docs/l10n.rst

```diff
@@ -9,17 +9,421 @@ Localization
+    <head>
+        <title>{{ ftl('example-page-title') }}</title>
+    </head>
+    <body>
+        <h1>{{ ftl('example-page-title') }}</h1>
+        <p>{{ ftl('example-page-desc') }}</p>
+        <footer>
+            <p>{{ ftl('example-footer') }}</p>
+        </footer>
+    </body>
+    </html>
+
+.. _Fluent file syntax: https://projectfluent.org/fluent/guide/
+
+FTL String IDs
+~~~~~~~~~~~~~~
+
+Our convention for string ID creation is the following:
+
+1. String IDs should be all lower-case alphanumeric characters.
+2. Words should be separated with hyphens.
+3. IDs should be prefixed with the name of the template file (e.g. ``firefox-new-skyline`` for ``firefox-new-skyline.html``)
+4. If you need to create a new string for the same place on a page and to transition to it as it is translated, you can
+   add a version suffix to the string ID: e.g. ``firefox-new-skyline-main-page-title-v2``.
+4. The ID should be as descriptive as possible to make sense to the developer, but could be anything as long as it adheres
+   to the rules above.
+
+The ``ftl`` helper function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``ftl()`` function takes a string ID and returns the string in the current language,
+or simplified english if the string isn't translated. If you'd like to use a different
+string ID in the case that the primary one isn't translated you can specify that like this:
+
+.. code-block:: python
+
+    ftl('primary-string-id', fallback='fallback-string-id')
+
+When a fallback is specified it will be used only if the primary isn't translated in the current
+locale. English locales (e.g. en-US, en-GB) will never use the fallback and will print the simplified
+english version of the primary string if not overridden in the more specific locale.
+
+You can also pass in replacement variables into the ``ftl()`` function for use with `fluent variables`_.
+If you had a variable in your fluent file like this:
+
+.. code-block:: fluent
+
+    welcome = Welcome, { $user }!
+
+You could use that in a template like this:
+
+.. code-block:: jinja
+
+    <h2>{{ ftl('welcome', user='Dude') }}<h2>
+
+For our purposes these are mostly useful for things that can change, but which shouldn't involve
+retranslation of a string (e.g. URLs or email addresses).
+
+This helper is available in Jinja templates and Python code in views. For use in a view you should
+always call it in the view itself:
+
+.. code-block:: python
+
+    # views.py
+    from lib.l10n_utils import render
+    from lib.l10n_utils.fluent import ftl
+
+    def about_view(request):
+        hello_string = ftl('about-hello')
+        render(request, 'about.html', {'hello': hello_string}, ftl_files='about')
+
+If you need to use this string in a view, but define it outside of the view itself, you can use the
+``ftl_lazy`` variant which will delay evaluation until render time. This is mostly useful for defining
+messages shared among several views in constants in a ``views.py`` or ``models.py`` file.
```

Comment:
Should we prefer a shared fluent file for these? Or at least mention that that's a perfectly fine option?